cluster/registry.go

```diff
@@ -13,16 +13,21 @@ import (
 
 const servicesPrefix = "services"
 
+type Registry interface {
+	Register(ctx context.Context, serviceName, nodeName, host string, port int) error
+	Services(ctx context.Context) (map[string][]Node, error)
+}
+
 type Node struct {
 	Address string `json:"address"`
 	Port    int    `json:"port"`
 }
 
-type ServiceRegistry struct {
+type serviceRegistry struct {
```

Comment:
do you guys think it's necessary to name this `etcdRegistry` instead? since we can support different stores now?

Reply:
I'm fine with etcdRegistry but why did you make this lower case?

Reply:
I didn't see a need to expose the underlying implimentation of the registry with the introduction of the interface

Reply:
right mb, i completely forgot about how golang uses caps for private/public